sycl/source/detail/sycl_mem_obj_t.cpp

```diff
@@ -91,7 +91,7 @@ void SYCLMemObjT::updateHostMemory() {
   // If we're attached to a memory record, process the deletion of the memory
   // record. We may get detached before we do this.
   if (MRecord)
-    Scheduler::getInstance().removeMemoryObject(this);
+    assert(Scheduler::getInstance().removeMemoryObject(this));
```

Comment:
Please, do not put `assert` around this call because in this case `removeMemoryObject` is not called in the build without asserts.

Reply:
fixed in https://github.com/intel/llvm/pull/6837/commits/71e9048af88ddcc853968cf9bbaa050f63ad323e